membrane_curvature/lib/mods.py

```diff
@@ -158,38 +52,35 @@ def core_fast_leaflet(z_Ref, leaflet, traj, jump, n_cells, lipid_types,
 
     """
 
-    grid_count = np.zeros([n_cells, n_cells])
+    grid_count_frames = np.zeros([n_cells, n_cells])
 
-    for frame in range(0, traj.n_frames, jump):
-        grid_1 = np.zeros([n_cells, n_cells])
-        grid_2 = np.zeros([n_cells, n_cells])
+    for ts in universe.trajectory:
 
-        factor = np.float32(n_cells / max_width)
-
-        for lipid_type in lipid_types:
+        grid_z_coordinates = np.zeros([n_cells, n_cells])
+        grid_norm_unit = np.zeros([n_cells, n_cells])
 
-            for bead in lipid_ref[leaflet][lipid_type]:
+        factor = np.float32(n_cells / max_width)
 
-                x, y, z = traj.xyz[frame, bead, :]
+        for bead in selection:
+            x, y, z = bead.position/10
 
-                l = int(abs(x) * factor)
-                m = int(abs(y) * factor)
+            l, m = grid_map((x, y), factor)
 
-                try:
-                    grid_1[l, m] += z
-                    grid_2[l, m] += 1
+            try:
+                grid_z_coordinates[l, m] += z
+                grid_norm_unit[l, m] += 1
 
-                except BaseException:
-                    pass
+            except ValueError:
+                print("Atom outside grid boundaries. Skipping atom {}".format(bead))
 
         for i, j in it.product(range(n_cells), range(n_cells)):
-            if grid_2[i, j] > 0:
-                z_Ref[i, j] += grid_1[i, j] / grid_2[i, j]
-                grid_count[i, j] += 1
+            if grid_norm_unit[i, j] > 0:
+                z_Ref[i, j] += grid_z_coordinates[i, j] / grid_norm_unit[i, j]
+                grid_count_frames[i, j] += 1
 
     for i, j in it.product(range(n_cells), range(n_cells)):
-        if grid_count[i, j] > 0:
-            z_Ref[i, j] /= grid_count[i, j]
+        if grid_count_frames[i, j] > 0:
+            z_Ref[i, j] /= grid_count_frames[i, j]
```

Comment:
I see that you're first creating an average surface before computing the curvature. On a scientific level, maybe we should consider treating each surface for each frame separately? Otherwise, if you have a membrane that undulates such that a patch has ~50% positive curvature and 50% negative curvature over the frames, the Gaussian curvature is ultimately computed on a flat surface?